src/lambda_streams/Animals.java

```diff
@@ -26,36 +28,110 @@ public static void main(String[] args) {
-
-        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
-        return instructions;
+        //reverse the order of the animals in the animal list.  If the booleaen parameter is true, reverse the static
+        // animals array list by mutating the array.  if the mutate boolean is false, flip a 'copy' of the animals
+        // list, then return that list of flipped animals, WITHOUT mutating the static animals array. return the
+        // flipped list in both cases.
+
+//        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals",
+//                "only","if","flag","true");
+//        return instructions;
+
+        if(mutate){
+            List <String> newAnimals = new ArrayList<String>(animals);
+            animals.clear();
+            newAnimals.stream()
+                    .collect(Collectors.toCollection(ArrayDeque::new))
+                    .descendingIterator()
+                    .forEachRemaining(x ->animals.add(x));
+            return animals;
+        }else{
+            List <String> newAnimals = new ArrayList<String>();
+            animals.stream()
+                    .collect(Collectors.toCollection(ArrayDeque::new))
+                    .descendingIterator()
+                    .forEachRemaining(x ->newAnimals.add(x));
+
+            return newAnimals;
+        }
+        //return animals;
     }
 
     static List<String> sortAnimals(boolean mutate) {
-    //sort the animals in alphabetical order.  If the booleaen parameter is true, mutating the animals list.  if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
-    return animals;
+    //sort the animals in alphabetical order.  If the boolean parameter is true, mutating the animals list.
+        // if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted
+        // animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
+        if(mutate){
+            List<String> sortedList = animals.stream()
+                    .sorted(Comparator.naturalOrder()).collect(Collectors.toList());
+            return sortedList;
+        }else{
+            List<String> newList = new ArrayList<String>(animals);
+            List <String> sortedClone = newList.stream()
+                    .sorted(Comparator.naturalOrder()).collect(Collectors.toList());
+            return sortedClone;
+        }
+
+    //return animals;
     }
```

Comment:
The expectation was to see lambdas created replacing each method, then the methods to be refactored to take those lambdas as an argument (e.g. capFirst(List<String>, boolean mutable, [LambdaType] lambda).